src/compas_model/elements/element.py

```diff
@@ -1138,86 +859,76 @@ def copy(self):
+        """Transforms all the attrbutes of the class.
 
         Parameters
         ----------
-        transformation : :class:`~compas.geometry.Transformation`
+        transformation : :class:`compas.geometry.Transformation`
             The transformation to be applied to the Element's geometry and frames.
 
         Returns
         -------
         None
 
         """
-        # transorm the geometry
-        self.frame.transform(transformation)
-        self.frame_global.transform(transformation)
 
-        for i in range(len(self.geometry_simplified)):
-            self.geometry_simplified[i].transform(transformation)
-
-        for i in range(len(self.geometry)):
-            self.geometry[i].transform(transformation)
-
-        # recompute the bounding-box
-        self._aabb.clear()
-        self.aabb()
+        # --------------------------------------------------------------------------
+        # main constructor properties
+        # --------------------------------------------------------------------------
+        self.frame.transform(transformation)
+        [g.transform(transformation) for g in self.geometry_simplified]
+        [g.transform(transformation) for g in self.geometry]
 
-        # transform the oobb and convex-hull
-        self.oobb()
-        for i in range(len(self._oobb)):
-            p = Point(self._oobb[i][0], self._oobb[i][1], self._oobb[i][2]).transformed(
-                transformation
-            )
-            self._oobb[i] = [p.x, p.y, p.z]
+        # --------------------------------------------------------------------------
+        # attributes that are dependent on a user given specifc data or geometry
+        # --------------------------------------------------------------------------
+        self.frame_global.transform(transformation)
 
-        if self.convex_hull:
-            if self.convex_hull.number_of_vertices() > 0:
-                self.convex_hull.transform(transformation)
+        self.compute_aabb()
+        self.compute_oobb()
 
     def transformed(self, transformation):
-        """Creates a transformed copy of the Element.
+        """Creates a transformed copy of the class.
 
         Parameters
         ----------
-        transformation : :class:`~compas.geometry.Transformation`:
+        transformation : :class:`compas.geometry.Transformation`:
             The transformation to be applied to the copy of an element.
 
         Returns
         -------
-        :class:`~compas_model.elements.Element`
+        :class:`compas_model.elements.Element`
             A new instance of the Element with the specified transformation applied.
 
         """
         new_instance = self.copy()
         new_instance.transform(transformation)
         return new_instance
 
-    def transform_to_frame(self, frame):
-        """Applies frame_to_frame transformation to the geometry , local frame, and global frame of the Element.
+    def transform_to_frame(self, target_frame):
```

Comment:
remove

Reply:
removed all the not needed transformations